Services/GEV/Desktop/classes/class.gevMyCoursesTableGUI.php

```diff
@@ -172,9 +170,18 @@ protected function fillRow($a_set) {
 		$this->tpl->setVariable("TARGET_GROUP", $a_set["target_group"]);
 		$this->tpl->setVariable("GOALS", $a_set["goals"]);
 		$this->tpl->setVariable("CONTENTS", $a_set["content"]);
-		$this->tpl->setVariable("CRS_LINK", gevCourseUtils::getLinkTo($a_set["obj_id"]));
 
-		
+		if($a_set["type"] == "Selbstlernkurs") {
+			$ref_id = gevObjectUtils::getRefId($a_set["obj_id"]);
+			$this->gCtrl->setParameterByClass("ilobjcoursegui", "ref_id", $ref_id);
+			$crs_link = $this->gCtrl->getLinkTargetByClass("ilobjcoursegui", "view");
+			$this->gCtrl->clearParametersByClass("ilobjcoursegui");
```

Comment:
I think this is not necessary, we will do this switch in the ilCourseGUI anyway, right?